### ISSUE TITLE
Replace inconsistent shadow styles with new shadow tokens

### ISSUE THREAD

**Issue by rmstinson** [Esri/calcite-design-system#10050]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

A handful of Calcite components use a different shadow style than the tailwind style `shadow.2`. Most of them currently use the style "0 0 16px 0 rgba(0, 0, 0, 0.16)." Some are inheriting the style from the classes "calcite-floating-ui-anim" + "calcite-floating-ui-anim--active." Those components' shadow styles need to be replaced with `shadow.2` in the code.

### Acceptance Criteria
- [x] Ensure the opacities used in shadow tokens are set in core
    - [x] 0.04
    - [x] 0.08
    - [x] 0.10
    - [x] 0.12
    - [x] 0.16

<details>
<summary>Some criteria moved to #14335</summary>

> - [ ] Add new semantic black color tokens for use with shadows:
> 
>     - `color.shadow.1`
>         - light mode: rgba({core.color.neutral.blk-240}, {core.opacity.4})
>         - dark mode: rgba({core.color.neutral.blk-240}, {core.opacity.12})
>     - `color.shadow.2`
>         - light mode: rgba({core.color.neutral.blk-240}, {core.opacity.8})
>         - darkmode: rgba({core.color.neutral.blk-240}, {core.opacity.12})
>     - `color.shadow.3`
>         - light mode: rgba({core.color.neutral.blk-240}, {core.opacity.10})
>         - dark mode: rgba({core.color.neutral.blk-240}, {core.opacity.16})
> 
> - [ ] Update semantic shadow tokens:
> 
>     - `shadow.sm` = box-shadow: 0px 2px 8px 0px {semantic.color.shadow.3}, 0px 2px 8px 0px {semantic.color.shadow.1};
>     - `shadow.md` = box-shadow: 0px 6px 20px -4px {semantic.color.shadow.3}, 0px 4px 12px -2px {semantic.color.shadow.2};
</details>

- [x] Update semantic shadow tokens:
    - `shadow.sm` = box-shadow: 0px 2px 8px 0px rgba({core.color.neutral.blk-240}, {core.opacity.10}), 0px 2px 8px 0px rgba({core.color.neutral.blk-240}, {core.opacity.4});
    - `shadow.md` = box-shadow: 0px 6px 20px -4px rgba({core.color.neutral.blk-240}, {core.opacity.10}), 0px 4px 12px -2px rgba({core.color.neutral.blk-240}, {core.opacity.8});
- [x] Apply shadow tokens as listed below:
- **`shadow.sm`**
  - ~Notice~ (no longer needed after redesign)
  - Sheet
  - Shell Panel `display-mode=“float-all” | “float-content” | “overlay”`
- **`shadow.md`**
  - Action Menu (renders a Popover)
  - Action Pad
  - Alert
  - Autocomplete
  - Dialog
  - Dropdown
  - FAB
  - Combobox / Input Time Zone
  - Input Date Picker
  - Input Time Picker
  - Menu Item (horizontal submenus)
  - Popover
  - Tooltip

### Relevant Info

[Figma spec reference](https://www.figma.com/design/quzdR1gCdG1wiZg4Jje8MJ/Consistent-Floating-UI-and-Shadows?node-id=301-305&m=dev&focus-id=1612-23045)

![Image](https://github.com/user-attachments/assets/32086d3e-39e5-4a20-afe5-2c96acf3b184)
![Image](https://github.com/user-attachments/assets/81bf56ea-96f6-461f-bcd0-bab8bf5e7655)

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [x] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Online

**monday.com sync:** #9365395346

**Comment by macandcheese** [comment]
Note: Shell Panel also has similar bespoke values for "directional" shadows based on "overlay" `displayMode` and Shell Panel placement.

**Comment by ashetland** [comment]
@anveshmekala Is that box-shadow needed? Button also uses an outset focus and does not do this.

**Comment by alisonailea** [comment]
> Updated issue title and description and marking `ready for dev`. cc [@matgalla](https://github.com/matgalla) [@alisonailea](https://github.com/alisonailea)

Updated issue description with additional Token Studio references

**Comment by ashetland** [comment]
@geospatialem, I'm assigning this to @matgalla for the initial tokens effort. The [last item](https://github.com/Esri/calcite-design-system/issues/10050#:~:text=Apply%20shadow%20tokens%20as%20listed%20in%20Which%20Component%20section%20below.) will need to be picked up once Matt's work is complete.

**Comment by matgalla** [comment]
Reopening after being closed due to unintentional automation

**Comment by matgalla** [comment]
[style-dictionary v5.3.0](https://github.com/style-dictionary/style-dictionary/blob/main/CHANGELOG.md?utm_source=chatgpt.com#530) and [v5.3.1](https://github.com/style-dictionary/style-dictionary/blob/main/CHANGELOG.md?utm_source=chatgpt.com#531) may have some helpful updates needed to address the limitations we've been seeing with shadow tokens being unable to incorporate color tokens intended to switch between light/dark modes. cc @jcfranco

**Comment by matgalla** [comment]
Due to tooling limitations, we are moving forward with a light-mode styling only solution in PR #14152 for the time being. Dark mode will not be getting the values spec'd here, but the update still improves the visibility in both modes vs their current versions so it should still be beneficial.

I've modified this issue's description to align with that plan and here's the follow-up issue that can be revisited when newer versions of Style Dictionary are released: #14335

**Comment by matgalla** [comment]
The shadow tokens have been updated (see previous comment) and will be applied to the specified components in the upcoming May milestone

**Comment by DitwanP** [comment]
🍠✨ Verified with version `5.1.0-next.60` and compared to `5.0.2` using doc site samples.

New shadow values:

<img width="775" height="51" alt="Image" src="https://github.com/user-attachments/assets/98b2c04a-fd5d-44d9-89d8-dd7daed32314" />

Example of component with new shadow:
Menu item before:

<img width="451" height="315" alt="Image" src="https://github.com/user-attachments/assets/9f31675f-74ea-4cf3-8e8b-40c7d38e95da" />

Menu Item after:

<img width="421" height="285" alt="Image" src="https://github.com/user-attachments/assets/2876e6e6-1ef3-4f6f-904f-b3f2c0b44490" />
